src/main/java/com/snowflake/kafka/connector/internal/streaming/schemaevolution/iceberg/IcebergColumnTypeMapper.java

```diff
@@ -101,9 +112,15 @@ public String mapToColumnType(Schema.Type kafkaType, String schemaName) {
           return "BINARY";
         }
       case ARRAY:
+        return "ARRAY";
+      case STRUCT:
+        return "OBJECT";
+      case MAP:
+        return "MAP";
       default:
-        // MAP and STRUCT will go here
-        throw new IllegalArgumentException("Arrays, struct and map not supported!");
+        // todo try to throw info about a whole record - this is pure
+        throw new IllegalArgumentException(
```

Comment:
nit: I think we could introduce a new 5xxx error code (like ERROR_5001).

Reply:
Done.